### PR TITLE
Remove the boolean type

### LINKED ISSUE #32: boolean_value_expression still in BNF
The BNF still mentions boolean_value_expression and all that's below it, including boolean_function, which isn't even defined.  I thought we wanted to drop all that?

### PR DESCRIPTION
**Pull request by gmantele**
Remove the boolean type for now, because of the grammar complexity it is introducing.
It will be added in a next version of ADQL after more thoughts (and probably with a different grammar syntax...PEG).

Resolves #32

### CODE REVIEW

**Comment by gmantele** [comment]
As announced about a month ago, and in absence of opposition, the boolean datatype is dropped from ADQL-2.1.

File: remove-boolean.patch [hunk 1]
```diff
@@ -0,0 +1,145 @@
+diff --git a/ADQL.tex b/ADQL.tex
```
Comment (jd-au) [inline remove-boolean.patch:1]: Was this file intended to be committed?

Comment (gmantele) [reply remove-boolean.patch:1]: Yes, it was. I just want to keep somewhere what was did to support boolean constraint. So, yes, I could use the Git history for that, but I would have to browse/search in the history. With this easy and simple patch I do not have to do so. But if you think it is useless, I keep it only locally on my computer.

**Review by jd-au — APPROVED** [review]
